### ISSUE TITLE
Hot Module Replacement

### ISSUE THREAD

**Issue by albertromkes** [pauldotknopf/react-dot-net#8]
Is it possible to use the Hot Module Replacement plugin? I've played around with it in the dev.webpack.config file (and start it with 'webpack-dev-server --config dev.webpack.config') but I can't get it to work. 

I also updated the `configureStore` method like this:

```
export default function configureStore(initialState, history, client) {
  const enhancer = compose(
    applyMiddleware(thunk),
    applyMiddleware(routerMiddleware(history)),
    applyMiddleware(createMiddleware(client)),
    devTools
  )(createStore);

  const finalEnhancer = enhancer(reducer, initialState);
  if (module.hot) {
    // Enable Webpack hot module replacement for reducers
    module.hot.accept('./reducer', () => {
      const nextRootReducer = require('./reducer');
      finalEnhancer.replaceReducer(nextRootReducer);
    });
  }
  // return enhancer(reducer, initialState);
  return finalEnhancer;
}
```

Any clues?

**Comment by pauldotknopf** [comment]
This is something I have been meaning to look into. It should be possible.

[Html.js](https://github.com/pauldotknopf/react-aspnet-boilerplate/blob/831bca5/src/React/Scripts/helpers/Html.js#L44) needs to be update to point at the separate instance of the webpack dev server. The application would be hosted on `http://localhost:80/`, while the scripts load from `http://localhost:81:/` (webpack dev server).

The server generated scripts will not support hot-reloading, since it doesn't really apply. However, there is a similar mechanism. run `gulp watch`, and as you change your files, the `server.generated.js` will get updated. The application internally monitors this file, and refreshes if it senses that it changed.

Once you get `client.generated.js` served up from webpack dev server, you should be golden from there. You will probally have to update some code (as you have found) because I initially didn't plan on supporting it. I'm open for some PRs though ;)

**Comment by amsterdamharu** [comment]
I was trying to use webpack-dev-server with a .net application and most of it works by starting 2 servers (the .net server to serve the content/data-endpoints and webpack-dev-server to serve the bundle with hot loading).

Using the following repo has hot loading, redux and and the redux dev tool:

https://github.com/gaearon/redux-devtools.git

Then I use webpack-dev-server proxy settings to have any missing content proxy to my .net application. This would work fine but the proxy does not seem to support ntlm (the .net application uses ntlm) and I am unable to mock/set a domain user in my dev environment (IIS Express/.net app).

Maybe you have solved this issue in a different way and I would curious to know how:
http://stackoverflow.com/questions/37928251/authenticate-net-web-application-as-specific-user

**Comment by pauldotknopf** [comment]
@amsterdamharu, this is the next issue/feature that I will be tackling. I will keep you updated. Probably won't get to it until this weekend.

**Comment by kosinw** [comment]
I am also having a similar issue where I have webpack-dev-server on one port and asp.net core running on the other, but whenever the hot module replacement api kicks in it searches for the update on the wrong port.

**Comment by georgiosd** [comment]
Any updates on this? Thank you :)

**Comment by pauldotknopf** [comment]
No, sorry. I have been swamped at my current position in lower level embedded Linux stuff. Not sure when I will have time to get back to this, ATM.

**Comment by georgiosd** [comment]
@pauldotknopf I can relate, no probs.

Would you be open to giving us a list of steps to implement this, in detail, and give some guidance with problems? I'd be open to sending over a PR if I can crack the nut.
